grails-app/views/stockMovement/_actions.gsp

```diff
@@ -39,27 +40,38 @@
                     </g:link>
                 </g:else>
             </div>
-            <g:isUserAdmin>
-                <g:if test="${(stockMovement?.isPending() || !stockMovement?.shipment?.currentStatus) && (isSameOrigin || !stockMovement?.origin?.isDepot())}">
-                    <hr/>
-                    <div class="action-menu-item">
-                        <g:if test="${stockMovement?.order}">
+            <g:if test="${(stockMovement?.isPending() || !stockMovement?.shipment?.currentStatus) && (isSameOrigin || !stockMovement?.origin?.isDepot())}">
+                <hr/>
+                <div class="action-menu-item">
+                    <g:if test="${stockMovement?.order}">
+                        <g:isUserAdmin>
                             <g:link class="button" controller="stockTransfer" action="remove" id="${stockMovement?.id}" params="[orderId: stockMovement?.order?.id]"
                                     onclick="return confirm('${warehouse.message(code: 'default.button.delete.confirm.message', default: 'Are you sure?')}');">
                                 <img src="${resource(dir: 'images/icons/silk', file: 'delete.png')}" />
                                 &nbsp;${warehouse.message(code: 'default.delete.label', args:[warehouse.message(code:'stockMovement.label')])}
                             </g:link>
-                        </g:if>
-                        <g:else>
+                        </g:isUserAdmin>
+                    </g:if>
+                    <g:elseif test="${stockMovement?.isElectronicType()}">
+                        <g:isUserAdmin>
+                            <g:link controller="stockRequest" action="remove" id="${stockMovement?.id}"
+                                    onclick="return confirm('${warehouse.message(code: 'default.button.delete.confirm.message', default: 'Are you sure?')}');">
+                                <img src="${resource(dir: 'images/icons/silk', file: 'delete.png')}" />
+                                &nbsp;${warehouse.message(code: 'default.delete.label', args:[warehouse.message(code:'stockMovement.label')])}
+                            </g:link>
+                        </g:isUserAdmin>
+                    </g:elseif>
+                    <g:else>
+                        <g:isUserInRole roles="[RoleType.ROLE_SUPERUSER, RoleType.ROLE_ADMIN, RoleType.ROLE_MANAGER, RoleType.ROLE_ASSISTANT]">
```

Comment:
yep, you're right, I had blind guess that it would be enough to write just 
```
<g:isUserInRole roles="[RoleType.ROLE_ASSISTANT]">
```
or create a new `<g:isUserAssistant>`, but I kept all roles thinking, that maybe the order might change in the future.
I have mixed feelings if the button should be visible for all, even if they don't have permissions - I mean, I followed the logic which was implemented in many places where we don't see something if we don't have enough permissions.
I wasn't feeling good with the fact, that when you don't have permission (blocked by "new" logic in RoleFilters) and you click the button, you are just redirected to blank page and you don't know what is going on (none message displays for the user), so as you probably say, it would be the best to modify `LinkTagLib`, so after clicking the button I'd have "Access Denied" (the button would be disabled), but I'll need to figure this out.

Reply:
> I have mixed feelings if the button should be visible for all

I've debated this with Kelsey a bunch and I think you're both probably right in this case. For users that will never have access to this button we should probably hide it. If the button is visible under certain situations (i.e. status changed) then it should probably be displayed but disabled with a message popup for why the user can't click the button.

So if we agree that the button needs to be hidden we can accomplish that either in the LinkTagLib itself or in a new wrapper taglib in the AuthTagLib. You could actually even keep using g:isUserInRole, but don't provide any roles just the controller and action that the . In the taglib you'd retrieve role(s) from the ConfigHelper.
```
<g:isUserInRole controller="${controllerName}" action="${actionName}">
```
or a new taglib if you don't want to mess around with (read: possibly break) the existing taglib that is used everywhere 
```
<g:isUserInMinimumRequiredRole controller="${controllerName}" action="${actionName}">
```
If there's some way to do this without passing the controller and action explicitly (i.e. do these get passed as attributes implicitly)?

```
def isUserInRole = { attrs, body ->
    if (session.user) {
        // Ask the config helper to return the default role(s). In most cases it'll be a 
        def defaultRoles = ConfigHelper.findDefaultRoles(attrs.controller, attrs.params)
        def requiredRoles = attrs.roles ?: defaultRoles
        def isUserInRole = userService.isUserInRole(session?.user?.id, requiredRoles)
        if (isUserInRole)
            out << body()
    }
}
```
Sources for UX discussion
https://ux.stackexchange.com/questions/107513/what-is-better-for-ux-hiding-or-disabling-irrelevant-buttons
https://ux.stackexchange.com/questions/24386/inactive-state-of-buttons-to-hide-or-not-to-hide-that-is-the-question
https://blogs.sap.com/2019/11/13/do-disabled-buttons-need-to-be-hidden/